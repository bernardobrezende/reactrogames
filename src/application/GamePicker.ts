import { Game } from '../model/Game';
import { getRandomIntBetween } from './Randomizer';

export default class GamePicker {

    games: Game[];

    constructor(games : Game[]) {
        this.games = games;
    }

    pick() : Game {
        const randomIndex = getRandomIntBetween(0, this.games?.length);
        return this.games[randomIndex];
    };
};
