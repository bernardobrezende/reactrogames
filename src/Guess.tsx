import PropTypes, { InferProps } from 'prop-types';
import Game from './Game';
import { Game as GameModel } from './model/Game';
import GamePicker from './application/GamePicker';

export default function Guess({ options } : { options: InferProps<typeof Guess.propTypes> }) {
    const selectedGame = new GamePicker((options as GameModel[])).pick();
    return <Game { ...selectedGame }/>;
};

Guess.propTypes = {
    options: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.string.isRequired,
        platform: PropTypes.string.isRequired,
        thumb: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired
    }))
};