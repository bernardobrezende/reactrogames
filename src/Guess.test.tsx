import { render } from '@testing-library/react';
import Guess from './Guess';
import * as randomizer from './application/Randomizer';
import { GAMES } from '../tests/fixtures/games';

describe('Guess', () => {
    it('should randomize game from several options', () => {
        randomizer.getRandomIntBetween = jest.fn(() => 1);
        const { container } = render(<Guess options={ GAMES } />);
        expect(container.firstChild).toMatchSnapshot();
    });
});