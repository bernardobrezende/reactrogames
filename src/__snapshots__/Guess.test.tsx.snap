// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guess should randomize game from several options 1`] = `
<div>
  <h1>
    Golden Axe
  </h1>
  <h2>
    SMS
  </h2>
  <a
    href="https://www.retrogames.cz/play_037-SegaMS.php"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="Golden Axe"
      src="/img/SMS-Golden_Axe.png"
    />
  </a>
</div>
`;
